### PR TITLE
brimcap load: index pcap

### LINKED ISSUE #6: brimcap load: index pcap
brimcap load should produce a pcap file for the analyzed pcap. The index name will be the name of the pcap file + `.idx`. The brimcap load command requires a `-root` arg that specifies where the pcap index file should exist. It should also include a symlink the indexed pcap file.

### PR DESCRIPTION
**Pull request by mattnibs**
Brimcap load should create an index and a symlink to the processed pcap,
written the BRIMCAP_ROOT.

Closes #6

### CODE REVIEW

File: cmd/brimcap/load/command.go [hunk 1]
```diff
@@ -102,6 +119,31 @@ func (c *Command) Exec(args []string) (err error) {
 	return err
 }
 
+func (c *Command) writeBrimcapRoot(path string, f *os.File, warner zbuf.Warner) error {
```
Comment (nwt) [inline cmd/brimcap/load/command.go:122]:
```suggestion
func (c *Command) writeBrimcapRoot(path string, rs io.ReadSeeker, warner zbuf.Warner) error {
```

File: cmd/brimcap/load/command.go [hunk 2]
```diff
@@ -102,6 +119,31 @@ func (c *Command) Exec(args []string) (err error) {
...
+
+	b, err := json.Marshal(index)
+	if err != nil {
+		return err
+	}
+
+	indexPath := filepath.Join(c.rootflags.Root, filepath.Base(path)+".idx")
+	if err := os.WriteFile(indexPath, b, 0644); err != nil {
+		return err
+	}
```
Comment (nwt) [inline cmd/brimcap/load/command.go:136]: Maybe use github/brimdata/zed/pkg/fs.MarshalJSONFiile?

File: pcap/pcapio/ngread.go [hunk 3]
```diff
@@ -67,13 +67,13 @@ func NewNgReader(r io.Reader) (*NgReader, error) {
...
-func (r *NgReader) SetWarningChan(c chan<- string) {
-	r.warningCh = c
+func (r *NgReader) SetWarningHandler(warner Warner) {
+	r.warner = warner
 }
 
 func (r *NgReader) warn(format string, a ...interface{}) {
-	if c := r.warningCh; c != nil {
-		c <- fmt.Sprintf(format, a...)
+	if c := r.warner; c != nil {
```
Comment (nwt) [inline pcap/pcapio/ngread.go:75]:
```suggestion
	if r.warner != nil {
```

File: cli/rootflags/flags.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
+package rootflags
+
+import (
+	"errors"
+	"flag"
+	"os"
+)
+
+type Flags struct {
```
Comment (nwt) [inline cli/rootflags/flags.go:9]: NIt: I don't think this deserves its own package under `cli`. I'd call it `RootFlags` and put it directly in `cli` instead.

**Review by nwt — APPROVED** [review]

File: cmd/brimcap/load/command.go (hunk 2 above)
Comment (mattnibs) [reply cmd/brimcap/load/command.go:136]: I find myself wanting to limiting the amount of things that brimcap depends on from zed, but perhaps I am being unnecessarily cautious.

File: cli/rootflags/flags.go (hunk 4 above)
Comment (mattnibs) [reply cli/rootflags/flags.go:9]: agreed.

File: cmd/brimcap/load/command.go (hunk 2 above)
Comment (mattnibs) [reply cmd/brimcap/load/command.go:135]: But then again fs.MarshalJSONFile does also update atomically which would prevent (unlikely) read races. Sure I'll do it!

**Pull request merged**